### DISCUSSION TITLE
Cannot upgrade to WebView2 XAML in Wpf app

### DISCUSSION THREAD

**Discussion by julianadormon** [MicrosoftEdge/WebView2Feedback#3913] [Q&A]
I have a hybrid WPF project which is running a Blazor app from a shared library.
Unfortunately, the XAML examples are lacking and I cannot figure out to upgrade the following code:

```xml
<Window x:Class="BlazorWpfApp.MainWindow"
        xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        xmlns:shared="clr-namespace:WebviewAppShared;assembly=WebviewAppShared"
        xmlns:local="clr-namespace:BlazorWpfApp"
        xmlns:blazor="clr-namespace:Microsoft.AspNetCore.Components.WebView.Wpf;assembly=Microsoft.AspNetCore.Components.WebView.Wpf"
        mc:Ignorable="d"
        Title="MainWindow" Height="450" Width="800">
    <DockPanel>
        <StackPanel Orientation="Horizontal" DockPanel.Dock="Top" Background="CadetBlue">
            <Label>This is native WPF UI</Label>
            <Button Click="Button_Click" Margin="4" Padding="4">Check counter</Button>
        </StackPanel>
        <blazor:BlazorWebView HostPage="wwwroot\index.html" Services="{StaticResource services}">
            <blazor:BlazorWebView.RootComponents>
                <blazor:RootComponent Selector="#app" ComponentType="{x:Type shared:App}" />
            </blazor:BlazorWebView.RootComponents>
        </blazor:BlazorWebView>
    </DockPanel>
</Window>
```

Source="wwwroot\index.html" is invalid and there are no "RootComponents" when using the new WebView2.

Is there a way to render the shared app in the WebView2 window? Thanks.

**Comment by champnic** [comment]
@eilon Any idea here?

**Comment by Eilon** [comment]
Hi @julianadormon , the `HostPage`, `Services`, and `RootComponents` properties are all specific to the `BlazorWebView` control, which is available in WPF, WinForms, and .NET MAUI projects. It internally creates a WebView2 control but it doesn't directly expose the WebView2's properties because most of those aren't applicable in Blazor scenarios.

Are you trying to use the WPF BlazorWebView to host a Blazor component in your WPF app, or are you specifically trying to use WebView2 to host arbitrary web/HTML content?

If you're looking for a WPF BlazorWebView example, you can find that here: https://github.com/dotnet/maui/tree/main/src/BlazorWebView/samples/BlazorWpfApp
